lang/cpp/cpp.py

```diff
@@ -20,12 +34,60 @@ def cpp_standard_type(m) -> str:
     return "std::" + m[1]
 
 
+@mod.capture(rule="{user.cpp_standard_prefix} {user.cpp_standard_function}")
+def cpp_standard_function(m) -> str:
+    # Discard the prefix word and insert properly prefixed function call
+    return "std::" + m[1]
+
+
+@mod.capture(rule="{user.cpp_standard_prefix} {user.cpp_standard_constant}")
+def cpp_standard_constant(m) -> str:
+    # Discard the prefix word and insert properly prefixed function call
+    return "std::" + m[1]
+
+
+@mod.capture(rule="{user.cpp_namespace}+")
+def cpp_namespace_list(m) -> str:
+    """List of namespaces followed by colons"""
+    return "::".join(m) + "::"
+
+
 @ctx.action_class("user")
 class UserActions:
     def code_get_operators() -> Operators:
         return operators
 
+    def code_insert_function(text: str, selection: str):
+        substitutions = {"1": text}
+        if selection:
+            substitutions["0"] = selection
+        actions.user.insert_snippet_by_name("functionCall", substitutions)
+
+
+@mod.capture(
+    rule="([<user.c_signed>] <user.c_types>) | <user.c_fixed_integer> | <user.cpp_standard_type>"
+)
+def code_type_raw(m) -> str:
+    return " ".join(list(m))
+
+
+@mod.capture(rule="<user.code_type_raw> [{user.cpp_pointers}+] <user.text>")
```

Comment:
We typically put something like formatting a variable in a command rather than a capture.